std/algorithm/iteration.d

```diff
@@ -4289,3 +4304,625 @@ unittest
          [1, 2, 0],
          [2, 1, 0]]));
 }
+
+/**
+Computes the $(WEB https://en.wikipedia.org/wiki/Binomial_coefficient, binomial coefficient)
+of $(D n,k).
+It is also known as "n choose k " or more formally as $(D n!/k!(n-k)).
+
+Params:
+    n = non negative integer
+    k = non negative integer, k <= n
```

Comment:
Thanks for the underscore hint - I added it to `n!/k!(n-k)`. 
I looked at a couple of examples in Phobos using preconditions and a typical value is `!empty` which isn't that helpful. I personally prefer to have extra conditions of variables at their explanation, so I left it. However I am open to more input on this ;-)
 
Btw ideally ddoc would parse the input contracts and display them.